### PR TITLE
fix sync bug for tsfiles's directory changed by vitural storage group

### PR DESCRIPTION
**Pull request by mychaow**
### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/sync/sender/manage/SyncFileManager.java [hunk 1]
```diff
@@ -47,33 +47,35 @@
...
    * id, inner value is the set of current sealed tsfiles.
+   * logicalSg -> virtualSg ->  <timeRangeId, files>
    */
-  private Map<String, Map<Long, Set<File>>> currentSealedLocalFilesMap;
+  private Map<String, Map<Long, Map<Long, Set<File>>>> currentSealedLocalFilesMap;
 
   /**
    * Key is storage group, value is all last local tsfiles in the storage group, which doesn't
    * contains those tsfiles which are not synced successfully. Inner key is time range id, inner
    * value is the set of last local tsfiles.
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/sync/sender/manage/SyncFileManager.java:64]: Update these comments?

File: server/src/main/java/org/apache/iotdb/db/sync/sender/manage/SyncFileManager.java [hunk 2]
```diff
@@ -102,35 +107,45 @@ public void getCurrentLocalFiles(String dataDir) {
...
-        try {
-          Long timeRangeId = Long.parseLong(timeRangeFolder.getName());
-          currentAllLocalFiles.get(sgFolder.getName()).putIfAbsent(timeRangeId, new HashSet<>());
-          File[] files = timeRangeFolder.listFiles();
-          Arrays.stream(files)
-              .forEach(file -> currentAllLocalFiles.get(sgFolder.getName()).get(timeRangeId)
+      for (File virtualSgFolder : sgFolder.listFiles()) {
+          try {
+            Long vgId = Long.parseLong(virtualSgFolder.getName());
+            allSGs.get(sgFolder.getName()).putIfAbsent(vgId, new HashSet<>());
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/sync/sender/manage/SyncFileManager.java:115]: The indent...

File: server/src/main/java/org/apache/iotdb/db/sync/sender/transfer/SyncClient.java [hunk 3]
```diff
@@ -470,17 +475,22 @@ public void sync() throws IOException {
           throw new SyncConnectionException("Unable to connect to receiver", e);
         }
         logger.info(
-            "Sync process starts to transfer data of storage group {}, it has {} time ranges.",
+            "Sync process starts to transfer data of storage group {}, it has {} virtual storage group.",
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/sync/sender/transfer/SyncClient.java:478]:
```suggestion
            "Sync process starts to transfer data of storage group {}, it has {} virtual storage groups.",
```

File: server/src/main/java/org/apache/iotdb/db/sync/sender/transfer/SyncClient.java [hunk 4]
```diff
@@ -515,9 +525,9 @@ public void syncDeletedFilesNameInOneGroup(String sgName, Long timeRangeId,
     logger.info("Start to sync names of deleted files in storage group {}", sgName);
     for (File file : deletedFilesName) {
       try {
-        if (serviceClient.syncDeletedFileName(file.getName()).code == SUCCESS_CODE) {
+        if (serviceClient.syncDeletedFileName(getFileNameWithSG(file)).code == SUCCESS_CODE) {
           logger.info("Receiver has received deleted file name {} successfully.", file.getName());
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/sync/sender/transfer/SyncClient.java:529]:
```suggestion
          logger.info("Receiver has received deleted file name {} successfully.", getFileNameWithSG(file));
```

**Review by HTHou — COMMENTED** [review]
LGTM

Just few minor problem. Please take a look. Thanks.

File: server/src/main/java/org/apache/iotdb/db/sync/sender/manage/ISyncFileManager.java [hunk 5]
```diff
@@ -55,13 +55,14 @@
    */
   void getValidFiles(String dataDir) throws IOException;
 
-  Map<String, Map<Long, Set<File>>> getCurrentSealedLocalFilesMap();
 
-  Map<String, Map<Long, Set<File>>> getLastLocalFilesMap();
+  Map<String, Map<Long, Map<Long, Set<File>>>> getCurrentSealedLocalFilesMap();
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/sync/sender/manage/ISyncFileManager.java:59]: this also needs comments... @return

**Review by jixuan1989 — COMMENTED** [review]
this is an important bugfix but there is something should be discussed.
Seems the client and the server have to have the same number of the virtual SG.
How can they guarantee that?
Add a check between them before the first sync?
By the way, is it responsible that the client and the server have the same number of the virtual SG?
(client may be lightweight which indicates it may have no resources to set up a large virtual SG) @SilverNarcissus

File: cross-tests/pom.xml [hunk 6]
```diff
@@ -84,4 +84,32 @@
...
+                <property>
+                    <name>skipUTs</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <properties>
+                <cross.ut.skip>true</cross.ut.skip>
+            </properties>
+        </profile>
+    </profiles>
```
Comment (SilverNarcissus) [inline cross-tests/pom.xml:114]: Could you explain the function of this part?

**Review by SilverNarcissus — COMMENTED** [review]
Nice work, only minor issues~ please have a look

**Review by qiaojialin — APPROVED** [review]

File: cross-tests/pom.xml (hunk 6 above)
Comment (qiaojialin) [reply cross-tests/pom.xml:114]: -DskipTests can not skip the test in cross-test module, so added this

File: server/src/main/java/org/apache/iotdb/db/sync/sender/manage/ISyncFileManager.java (hunk 5 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/sync/sender/manage/ISyncFileManager.java:62]: virtual sg configuration in sg level is needed. Then the receiver could follow the configuration of the sender and each sg should be synced by one sender.

**Pull request merged**